### ISSUE TITLE
(Feature) Save planet from domination missions

### ISSUE THREAD

**Issue by pksebben** [endless-sky/endless-sky#6508]
### Problem Description

Big, planet capping space battles are tons of fun, but they break a ton of stuff when you do 'em, because for some reason folks hate having to pay tribute.  Go figure.

So, I was reading through #4278, and it hit me that if you, as a swashbuckling space buccaneer, can demand tribute from planets, what's to keep other spacefaring swashbucklers from doing the same?  Wouldn't the denizens of such a planet want something done about that?

(I realize that this is sort of covered in the "hunt down X marauder missions", but stay with me, because this leads into what I think could be a truly rad mechanic)

### Related Issue Links

tangentially related to #4278, but barely

### Desired Solution

Planets / systems that (given certain conditions like a high enough combat rating) get 'occupied' by pirates, giving the player the opportunity to (through a mission, perhaps) liberate the planet in a "gimme tribute" style planetary defense battle.

Wouldn't have to apply to every planet, either.  Say the Ruchbah system periodically get co-opted by pirates and you get hired by the syndicate to liberate 'em.  

Planets under temporary pirate rule could behave like a pirate planet (pay us or don't land) and change the fleet spawns to increase pirate presence.  

### Alternative Approaches

To avoid a bunch of planets randomly and annoyingly becoming pirate worlds (say that the player is going for a different style of play than "blow up all the things"), this could be triggered only on acceptance of a mission.  That way the mechanic becomes opt-in.

**Comment by Zitchas** [comment]
Seems like a good idea. I could see having one or two random planets getting captured and then having missions to free them. Could be a good option for late-game missions.

That being said, I could also see this being a great plug-in. A wave of pirate aggression conquers a large number of planets, now the player can go free them in epic battles.

**Comment by pksebben** [comment]
I've been reading through the [mission guide on the wiki](https://github.com/endless-sky/endless-sky/wiki/CreatingMissions#triggers) -- is it possible using those mechanics to swap out planet governments, and replace them with the original gov on a tribute trigger?

**Comment by Amazinite** [comment]
Is something like this what you're looking for?
```
mission "pirates invade winter"
	landing
	invisible
	to offer
		# Necessary so that we know what government to change back to
		has "event: war begins"
		random < 1
	on offer
		event "pirates invade winter"
		fail

event "pirates invade planet"
	system "Yed Prior"
		government "Pirate"
		...
		# Change fleets here

mission "player retakes winter"
	landing
	invisible
	to offer
		has "tribute: Winter"
	on offer
		event "player retakes winter"
		fail

event "player retakes winter"
	system "Yed Prior"
		government "Free Worlds"
		...
		# Change fleets back
```

**Comment by pksebben** [comment]
it absolutely is.  Was this already implemented?

**Comment by Amazinite** [comment]
> ~~Was this already implemented?~~

No, that's just me whipping up the syntax for how you might go about implementing this idea for a single planet/system.

As far as the idea itself goes though, I'm not too keen on the idea of having planets swap to the pirate government. Important to note is that demanding tribute from planets is basically a minigame for players with nothing else to do or players who want to test their strength. It isn't really intended as "canon" gameplay, as the story doesn't react to the player dominating planets at all. So insofar as dominating planets is more an extra bit of content for late game players than it is a fully fleshed out system, I wouldn't expect pirates attempting to do the same to be very common. Pirates taking over planets seems to me to be far more fitting of a proper storyline than it does a random event that can occur.

**Comment by Amazinite** [comment]
Something else to note is that we already do have missions where the player helps defend systems from invading pirates: "[Southern, Northern, Core] Pirate Attack" in `jobs.txt` and "Pirate Occupation" in `south jobs.txt`.

**Comment by Amazinite** [comment]
I updated the template a bit to reset your tribute with Winter when the pirates invade; should you have already dominated Winter, we wouldn't want the invasion to immediately end the moment it started. This runs into an issue though where dominated planets can't be undominated through condition changes without reloading the save, so the player would need to relinquish tribute with Winter even though their tribute was already reduced to 0 before they can dominate Winter again and therefore end the invasion.

Alternatively, you could just have planets that the player has already dominated not be invaded by pirates, at least until we have some way to have missions/events cause the player to relinquish tribute from a planet.

**Comment by pksebben** [comment]
Part of the motivation behind this concept is that it is indeed a late-late-game sort of extension of the challenges presented.  What if there were a storyline that kicked off a repeating mission cron for like, a small handful of planets (managed such that they wouldn't clobber other story or missions - I could try and ensure that any undone story missions would preclude the offer).

What I'm picturing here lorewise is sort of like the wild geese a la nova.  You sign on and after a few missions to intro the mechanic you can grab em off the bbs.

|  the player would need to relinquish tribute with Winter even though their tribute was already reduced to 0 before they can dominate Winter again and therefore end the invasion.

I knew there was gonna be a gotcha like this!  Hadn't thought of that one though.  Wouldn't it be possible to block the mission if you have the planet dominated already?  Would also fit the lore side, I wouldn't hire a jerk I was paying 'protection money' to if I could avoid it...

**Comment by Amazinite** [comment]
> Wouldn't it be possible to block the mission if you have the planet dominated already?

Yup. In the mission that would cause Winter to be occupied, you'd have the following condition in the `to offer`:

```
to offer
	not "tribute: Winter"
	...
```

**Comment by movingpictures** [comment]
Related idea: what if other, just-as-savage fleets come in and dominate a planet the player has already dominated?

**Comment by Dart52000** [comment]
> Related idea: what if other, just-as-savage fleets come in and dominate a planet the player has already dominated?

as mentioned, in that case the planet would ask for help from someone they actually liked instead of someone who previously attacked their planet

**Comment by movingpictures** [comment]
You're mis-reading: I am referring to the idea of a random NPC taking the
planet away, meaning the player loses revenue.  Just because. The new
"owner" may also bolster defences, making re-domination a major challenge.

**Comment by Amazinite** [comment]
Closing since, as mentioned by my previous comments, I don't see us adding this to vanilla, but one is free to use the mission examples I provided to do this on their own as a plugin. In terms of vanilla, events such as this are typically more fit to be tied to a storyline rather than being something that just randomly happens form time to time.
